test/test_encryption.py

```diff
@@ -578,120 +598,133 @@ def create_key_vault(vault, *data_keys):
+            "db.coll": {
+                "bsonType": "object",
+                "properties": {
+                    "encrypted_placeholder": {
+                        "encrypt": {
+                            "keyId": "/placeholder",
+                            "bsonType": "string",
+                            "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+                        }
+                    }
                 }
-              }
             }
-          }
         }
         opts = AutoEncryptionOpts(
-            self.kms_providers(), 'keyvault.datakeys', schema_map=schemas)
-        client_encrypted = rs_or_single_client(
+            cls.KMS_PROVIDERS, 'keyvault.datakeys', schema_map=schemas)
+        cls.client_encrypted = rs_or_single_client(
             auto_encryption_opts=opts, uuidRepresentation='standard')
-        self.addCleanup(client_encrypted.close)
+        cls.client_encryption = ClientEncryption(
+            cls.KMS_PROVIDERS, 'keyvault.datakeys', cls.client, OPTS)
 
-        client_encryption = ClientEncryption(
-            self.kms_providers(), 'keyvault.datakeys', client, OPTS)
-        self.addCleanup(client_encryption.close)
+    @classmethod
+    def tearDownClass(cls):
+        cls.client.close()
+        cls.vault.drop()
+        cls.client_encrypted.close()
+        cls.client_encryption.close()
 
-        # Local create data key.
-        listener.reset()
-        local_datakey_id = client_encryption.create_data_key(
-            'local', key_alt_names=['local_altname'])
-        self.assertBinaryUUID(local_datakey_id)
-        cmd = listener.results['started'][-1]
-        self.assertEqual('insert', cmd.command_name)
-        self.assertEqual({'w': 'majority'}, cmd.command.get('writeConcern'))
-        docs = list(vault.find({'_id': local_datakey_id}))
-        self.assertEqual(len(docs), 1)
-        self.assertEqual(docs[0]['masterKey']['provider'], 'local')
+    def setUp(self):
+        self.listener.reset()
 
-        # Local encrypt by key_id.
-        local_encrypted = client_encryption.encrypt(
-            'hello local',
-            Algorithm.AEAD_AES_256_CBC_HMAC_SHA_512_Deterministic,
-            key_id=local_datakey_id)
-        self.assertEncrypted(local_encrypted)
-        client_encrypted.db.coll.insert_one(
-            {'_id': 'local', 'value': local_encrypted})
-        doc_decrypted = client_encrypted.db.coll.find_one({'_id': 'local'})
-        self.assertEqual(doc_decrypted['value'], 'hello local')
-
-        # Local encrypt by key_alt_name.
-        local_encrypted_altname = client_encryption.encrypt(
-            'hello local',
-            Algorithm.AEAD_AES_256_CBC_HMAC_SHA_512_Deterministic,
-            key_alt_name='local_altname')
-        self.assertEqual(local_encrypted_altname, local_encrypted)
+    def run_test(self, provider_name):
+        # Create data key.
+        master_key = self.MASTER_KEYS[provider_name]
+        if master_key is not None:
```

Comment:
Aren't both calls to `create_data_key` equivalent here since `master_key=None` is valid?

Reply:
Good point.